src/lib/axios.ts

```diff
@@ -49,13 +59,35 @@ export class AxiosManager {
   }
 
   private async rejectResponseInterceptor(
-    error: AxiosError<{ errors: string[] }>
+    error: AxiosError<{ message: string }>
   ) {
     const originalRequest = error.config
+    const errorMessage = error?.response?.data.message
+
     if (!originalRequest) {
       return Promise.reject(error)
     }
-    //TODO: handle error when accessToken invalid
+
+    if (
+      errorMessage?.includes('token is expired') ||
+      errorMessage?.includes('token is invalid')
+    ) {
+      try {
+        const { data } = await this.axiosRefresh.post<
+          any,
+          { data: CreateAuthRefreshTokenReesponse }
+        >(`v2/corporate/auth/refresh`)
+
+        store.dispatch(validAuth(data.data))
+        localStorage.setItem('accessToken', data.data.token)
```

Comment:
```suggestion
        store.dispatch(validAuth(data.data))
        localStorage.setItem('accessToken', data.data.token)
        return this.axios(originalRequest)
```

tambahin ini ya, biar kalo sukses dapet accessToken baru dia run ulang original requestnya

Reply:
ini kayanya udh dibantu react query deh tanpa di rerun udh otomatis rerun axiosnya, tadi gw coba sama get subscription token

Reply:
ok menarik jg react query

Reply:
eh tapi bukannya itu block if nya kalo gak direturn bakal ke bawah ketemu `Promise.reject(error)` ya?

Reply:
![image](https://github.com/PurwadhikaDev/purwadhika-corporate-website/assets/94028480/4f656f6a-4c8a-43f8-8956-462e64e8376e)
iya promise yang pertama bakal ke itung 401, abis itu react query rerequest lagi

Reply:
![image](https://github.com/PurwadhikaDev/purwadhika-corporate-website/assets/94028480/95fd6154-ebc3-4a2e-b71e-94cb4e530cf6)
tapi boleh deh kayanya biar axiosnya aja yang rerequest, in case ada yang gapake react query. flownya sama sih kalau return axios cuman pure axios yang ngejalanin rerequest bukan react querynya